### PR TITLE
Add MySQL Modeling Rules

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
- [x] Ready

## Related Issues
fixes: link to the issue

## Description
Adds the XSIAM modeling rules for MySQL.

## Minimum version of Cortex XSIAM
- [x] 3.3.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MySQLEnterprise/ModelingRules/MySQLEnterpriseEventCollector/MySQLEnterpriseEventCollector.xif [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[MODEL: dataset=mysql_enterprise_raw, model=Audit, content_id=***, version=***]
+| alter XDM.Audit.operation.type = json_extract_scalar(_raw_log, ""$.class"")
+| filter XDM.Audit.operation.type = ""audit""
+| alter XDM.Database.time = json_extract_scalar(_raw_log, ""$.timestamp""),
```
Comment (avidan-H) [inline Packs/MySQLEnterprise/ModelingRules/MySQLEnterpriseEventCollector/MySQLEnterpriseEventCollector.xif:4]: I think this is probably an error and it's supposed to be `XDM.Audit.event_timestamp`...

File: Packs/MySQLEnterprise/ModelingRules/MySQLEnterpriseEventCollector/MySQLEnterpriseEventCollector.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+name: MySQLEventCollector
+id: MySQLEventCollector
+fromversion: 3.3.0
```
Comment (daryakoval) [inline Packs/MySQLEnterprise/ModelingRules/MySQLEnterpriseEventCollector/MySQLEnterpriseEventCollector.yml:3]: Lets set this for debugging purpose and check if it causes the problem:
@yaakovi 

```suggestion
fromversion: 6.0.0
```

**Comment by avidan-H** [comment]
@evisochek can you review the updated modeling rules? I changed a lot of the fields to what I believe are the updated versions of the fields in the relevant models (e.g. Auth, Audit, Database) but I still think you should make sure I mapped to the correct things since some of them I was making assumptions. Also, is my [comment here](https://github.com/demisto/content/pull/19112#discussion_r878785936) correct?

File: Packs/MySQLEnterprise/ModelingRules/MySQLEnterpriseEventCollector/MySQLEnterpriseEventCollector.xif (hunk 1 above)
Comment (evisochek) [reply Packs/MySQLEnterprise/ModelingRules/MySQLEnterpriseEventCollector/MySQLEnterpriseEventCollector.xif:4]: Yes

**Review by evisochek — APPROVED** [review]

**Review by yaakovi — APPROVED** [review]

**Comment by avidan-H** [comment]
@anara123 @evisochek Since this pack's modeling rules are decoupled from the way the events are collected (i.e. it doesn't include an integration to fetch) do we need to include documentation here on how the event collection is configured or is this something that is known or documented on the side of xsiam platform?

**Comment by avidan-H** [comment]
@evisochek regarding the dataset, when we manually tested together we were testing against the `filestream_filestream_raw` dataset, but you mentioned that the dataset was liable to change to `<vendor>_<product>_raw` which is why here I have the dataset as `mysql_enterprise_raw` in the PR. Do we have information regarding when/if it will change? The dataset name mismatch is problematic right [now](https://jira-hq.paloaltonetworks.local/browse/XDR-56028). Also, since I don't know how the collection of these events logs are configured in the platform, it's unclear to me if the dataset name will always be the same or if it's something that the user will set at configuration time, in which case we would need to document somewhere that they need to set the dataset name to match the one used here in the modeling rules.

**Comment by evisochek** [comment]
> @evisochek regarding the dataset, when we manually tested together we were testing against the `filestream_filestream_raw` dataset, but you mentioned that the dataset was liable to change to `<vendor>_<product>_raw` which is why here I have the dataset as `mysql_enterprise_raw` in the PR. Do we have information regarding when/if it will change? The dataset name mismatch is problematic right [now](https://jira-hq.paloaltonetworks.local/browse/XDR-56028). Also, since I don't know how the collection of these events logs are configured in the platform, it's unclear to me if the dataset name will always be the same or if it's something that the user will set at configuration time, in which case we would need to document somewhere that they need to set the dataset name to match the one used here in the modeling rules.

Please see my reply in : https://jira-hq.paloaltonetworks.local/browse/XDR-52867

**Comment by avidan-H** [comment]
I tested it manually with an instance that had a dataset and the model fields mapped correctly. I was also successfully able to upload the pack to content's test instance 2 so I will merge upon a successful build unless there are any objections.

File: Packs/MySQLEnterprise/pack_metadata.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "MySQL Enterprise",
+    "description": "MySQL Enterprise Pack",
```
Comment (avidan-H) [inline Packs/MySQLEnterprise/pack_metadata.json:3]:
```suggestion
    "description": "The modeling rules for MySQL Enterprise event collectors.",
```

**Pull request merged**